src/model/Peptide.cpp

```diff
@@ -1137,6 +1137,35 @@ bool Peptide::lessThan(
   return (seq_one[pep_idx] < seq_two[pep_idx]);
 }
 
+bool Peptide::lessThanStr(
```

Comment:
Can you add a comment about what this function is supposed to do?  It seems convoluted.  I think if one peptide is a prefix of the other, then it returns true. Is that what is intended?